prow/cmd/deck/job_history.go

```diff
@@ -434,8 +435,11 @@ func getJobHistory(ctx context.Context, url *url.URL, cfg config.Getter, opener
 		tmpl.LatestLink = linkID(url, emptyID)
 	}
 
-	buildIDs, err := bucket.listBuildIDs(ctx, root)
-	if err != nil {
+	// Don't spend an unbound amount of time finding a potentially huge history
+	buildIDListCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
Is this best effort or will timeouts produce no results?

Reply:
Best effort, that is why the other change was required. It seems there is no way to limit the number of result, the timeout seems to be the only way to somehow limit this.